runtime/src/main/java/org/corfudb/runtime/collections/TransactionPoller.java

```diff
@@ -69,18 +83,12 @@ private void pollTxStream() {
  
         long lastReadAddress = streamContexts.get(0).getLastReadAddress();
 
-        StreamOptions options = StreamOptions.builder()
-                .ignoreTrimmed(true)
-                .cacheEntries(false)
-                .build();
-        IStreamView txStream = runtime.getStreamsView()
-                .get(ObjectsView.TRANSACTION_STREAM_ID, options);
         log.trace("Seeking txStream to {}", lastReadAddress + 1);
-        txStream.seek(lastReadAddress + 1);
+        txnStream.seek(lastReadAddress + 1);
         log.trace("txStream current global position after seeking {}, hasNext {}",
-                txStream.getCurrentGlobalPosition(), txStream.hasNext());
+                txnStream.getCurrentGlobalPosition(), txnStream.hasNext());
 
-        List<ILogData> updates = txStream.remaining();
+        List<ILogData> updates = txnStream.remaining();
```

Comment:
If the delta is huge, then this might create a memory issue. Why don't you just continuously call next?

That way you shift the memory issue with a latency issue, later you can introduce a remaining method to the stream that can consume the delta in chunks instead of constructing all the delta in memory first and then consuming it.

Reply:
This is a very good observation - in fact in some consumers the heap memory available is so small that its quite possible with large transactions reconstruction of the deltas cause OOM.
I think this warrants a simpler next() call based streaming. thanks!